coil-base/src/main/java/coil/request/RequestDelegate.kt

```diff
@@ -86,6 +86,11 @@ internal class ViewTargetRequestDelegate(
         lifecycle.removeObserver(this)
     }
 
+    override fun complete() {
+        if (target is LifecycleObserver) lifecycle.removeObserver(target)
+        lifecycle.removeObserver(this)
```

Comment:
We can't remove the `target` lifecycle observer here as it's used by `ImageViewTarget` (and other custom targets) to start/stop animations in onStart/onStop.